Restart dynamo analysis when we fail to tensorify away all symfloat inputs

bobrenjc93 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #140893
* #138922
* #138915
* __->__ #140346

Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

bobrenjc93 committed:
Restart dynamo analysis when we fail to tensorify away all symfloat inputs

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

bobrenjc93 commented on torch/fx/passes/_tensorify_python_scalars.py:
```diff
@@ -281,6 +283,21 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
+            # At this point we've lost the back pointer to
+            # what f_local this placeholder points to. Instead,
+            # we will rely on the index to specialize when we
+            # restart analysis.
```
This is probably the most controversial part of this PR. An alternative universe is to have AOTAutoGrad propogate placeholder names to this point and we can have hoist the specialization logic up to the wrap_symfloat level.

bobrenjc93 commented on torch/fx/passes/_tensorify_python_scalars.py:
```diff
@@ -281,6 +283,21 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
 
+    should_restart = False
+    for i, node in enumerate(graph.nodes):
+        if node.op == "placeholder" and len(node.users) == 0:
```
If we want to optimize for performance perhaps it make sense to break early once we iterate past the placeholders.

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


[ghstack-poisoned]

bobrenjc93 commented on torch/fx/passes/_tensorify_python_scalars.py:
```diff
@@ -281,6 +288,37 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
+            TensorifyState.specialize(name)
+            should_restart = True
+
+    # Sometimes by the time we get to tensorify, there have already been
```
This is the meat of the PR. Because by the time we reach tensorify we've lost all source/name information in the fx graph, I utilize shape_env to figure out which specializations have already happened.

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`


cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 commented on torch/fx/passes/_tensorify_python_scalars.py:
```diff
@@ -281,6 +307,48 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
+    #     item: "Sym(zf0)" = l_scale_factor_.item();  l_scale_factor_ = None
+    #     interpolate: "f16[2, 3, TruncToInt(4.0*zf0)][3*TruncToInt(4.0*zf0), TruncToInt(4.0*zf0), 1]cuda:0" = torch.nn.functional.interpolate(l_tensor_, size = None, scale_factor = item, mode = 'nearest-exact', align_corners = None, recompute_scale_factor = True);  l_tensor_ = item = None # noqa: B950
+    #     return (interpolate,)
+    for i, node in enumerate(graph.nodes):
```
I have an experimental improvement in https://github.com/pytorch/pytorch/pull/140829

I'll wait for CI to get green before squashing but the high level idea is to instead maintain a whitelist of symfloats we managed to tensorify away and then specialize anything not in the whitelist. Makes a lot of this logic much cleaner.

bobrenjc93 commented:
@bobrenjc93 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66032797).

bobrenjc93 commented:
@bobrenjc93 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66032986).

bobrenjc93 commented:
@bobrenjc93 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66032986).

bobrenjc93 commented:
@bobrenjc93 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66033375).

bobrenjc93 commented:
@bobrenjc93 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66033375).

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 commented on test/dynamo/test_unspec.py:
```diff
@@ -659,7 +659,7 @@ def fn(x, y):
             cnt = CompileCounterWithBackend("aot_eager")
             fn_opt = torch._dynamo.optimize(cnt)(fn)
-            x = torch.tensor(9.734375, dtype=dtype, requires_grad=True)
+            x = torch.randn(5, dtype=dtype, requires_grad=True)
```
I just changed this to be a non scalar tensor to make the test simpler, otherwise we would have different frame counts depending on if dtype == torch.float64. This is because were were assigning zf0 to 9.734375 and zf1 to the concrete value of y. So at the time of tensorify, we would tensorify away zf1 but not zf0 causing a restart.

I suppose this highlights one of the downsides of the current implementation where sometimes we will restart analysis when we don't really need to. I guess in the most ideal world at the time of tensorify we would know what was specialized before this point in time (eg. in arg_parser.h or in fake tensor prop). Then we could know which non tensorified away symfloats are okay.

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

ezyang commented on torch/_dynamo/convert_frame.py:
```diff
@@ -758,6 +765,8 @@ def log_bytecode(
                     unimplemented("100+ RestartAnalysis() calls")
             except exc.SkipFrame as e:
+                if not isinstance(e, exc.TensorifyScalarRestartAnalysis):
+                    TensorifyState.clear()
```
I'm not sure how to refactor this but having to manually clear the states here seems error prone (and I also remember stubbing my toe on this with compiler collectives, see https://github.com/pytorch/pytorch/pull/131983

ezyang commented on torch/_dynamo/symbolic_convert.py:
```diff
@@ -245,6 +246,22 @@ class DistributedState:
     all_states: Optional[List[LocalState]] = None
 
 
+class TensorifyState:
```
Docs. Esp what the str denotes here

ezyang commented on torch/fx/passes/_tensorify_python_scalars.py:
```diff
@@ -281,6 +306,27 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
+    if should_restart:
+        # Sledgehammer time. Restart dynamo analysis, keeping track of which input sources
+        # are no longer needed and should be specialized.
+        raise TensorifyScalarRestartAnalysis
```
Similar to https://github.com/pytorch/pytorch/pull/131983 need to clear speculation log

ezyang replied:
Not sure what happens if compiler collectives is going on at this point too

ezyang replied:
I'm looking for a comment that says something like "restarting analysis is necessary because we need to instruct Dynamo to NOT make these as inputs."

It really does seem like CUDA graph should be able to ignore inputs that don't get used inside the body though LOL

ezyang commented on torch/fx/passes/_tensorify_python_scalars.py:
```diff
@@ -109,6 +112,8 @@ def tensorify_python_scalars(
     tracer = fx.proxy.GraphAppendingTracer(graph)
     expr_to_sym_proxy: dict[sympy.Expr, MetaProxy] = {}
     expr_to_tensor_proxy: dict[sympy.Expr, MetaProxy] = {}
+    tensorified_symbols: set[str] = set()
```
Why not just store sympy.Symbol here, if it's symbols

ezyang commented on torch/fx/passes/_tensorify_python_scalars.py:
```diff
@@ -207,6 +211,24 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
 
+            # Specialize all dimensions that contain symfloats. Here's
+            # an example test that requires this:
+            # PYTORCH_OPINFO_SAMPLE_INPUT_INDEX=4 tlp python test/inductor/test_torchinductor_opinfo.py TestInductorOpInfoCUDA.test_comprehensive_nn_functional_interpolate_bicubic_cuda_float32 # noqa: B950
```
s/tlp//

ezyang replied:
This seems incredibly suspicious. You shouldn't ever have a SymFloat that comes out of tensor.shape....

bobrenjc93 replied:
IIRC it was something like Floor(zf0 * 1.5) or something like that. I can run the repro again to get the exact condition if you'd like.

ezyang replied:
Definitely a bug. FloorToInt should be used when it gets coerced to int, if that's the case.

ezyang commented on torch/fx/passes/_tensorify_python_scalars.py:
```diff
@@ -207,6 +211,24 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
+                for dim in val.shape:
+                    if isinstance(
+                        dim,
+                        (torch.SymFloat, torch.SymInt, torch.SymBool),
```
so it seems to me that the SymInt test here should suffice

ezyang commented on torch/fx/passes/_tensorify_python_scalars.py:
```diff
@@ -207,6 +211,24 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
+                                s, SymT.FLOAT
+                            ) and not TensorifyState.should_specialize(name):
+                                TensorifyState.specialize(name)
+                                should_restart = True
```
This is good, although it's probably worth elaborating in a little more detail why we choose to spec here. In principle, we could support float input that is used to do size compute. The problem is that we don't actually want to tensorify the compute in this case, which means need support for codegen all the float! (Maybe if someone comes to us with a legitimate use case, this restriction could be lifted, perhaps.)

ezyang commented on torch/fx/passes/_tensorify_python_scalars.py:
```diff
@@ -281,6 +306,27 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
+    # placeholder nodes no longer have a reference to their original
+    # symfloat and thus we need to deduce specializations have happend
+    # via shape_env.replacements. NB: there's an important invariant here
+    # that symfloats keep consistent names across restarts.
```
So what, you have to allocate the symfloat name even if you're not going to use it at all? Seems sus lol

ezyang replied:
Don't we associate every var with sources?

```
        self.var_to_sources: Dict[sympy.Symbol, List[Source]] = {}
```

So that's how you get sources.

ezyang commented on torch/_dynamo/output_graph.py:
```diff
@@ -1617,6 +1628,38 @@ def update_used_symbols(used_symbols, fake: Union[torch.SymInt, torch.Tensor]):
+        # save the state of what inputs we should have specialized in dynamo and
+        # restart analysis. This function incorporates this "view from the future"
+        # state and specializes inputs that we know we won't be able to tensorify
+        # away in the joint pass.
```
So explicitly say that in principle we shouldn't choke and so this is not necessary

ezyang commented on torch/_dynamo/output_graph.py:
```diff
@@ -1617,6 +1628,38 @@ def update_used_symbols(used_symbols, fake: Union[torch.SymInt, torch.Tensor]):
+        from torch._dynamo.symbolic_convert import TensorifyState
+
+        for node in self.graph.nodes:
+            example_value = node.meta.get("example_value")
```
I feel like grapharg should more directly get you to what you want here

ezyang reviewed (approved):
Looks good, please see comments

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Restart dynamo analysis when we fail to tensorify away all symfloat inputs"


Fixes a bunch of benchmarks that failed with cudagraph errors including `tlp python benchmarks/dynamo/timm_models.py --device cuda --inductor --accuracy --amp --training --only resmlp_12_224` when `specialize_float=False`

Also brings down number of overall failures (with keep-going) from 108 => 62. I'd estimate >80% of those 62 are wobbly expect tests. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 committed:
Update

[ghstack-poisoned]

bobrenjc93 commented:
@pytorchbot merge
